contracts/FeralfileArtworkV4.sol

```diff
@@ -10,60 +10,216 @@ import "./Authorizable.sol";
+    bool private _bridgeable;
+
+    // mintable
+    bool private _mintable = true;
 
     // token base URI
     string internal _tokenBaseURI;
 
     // contract URI
     string private _contractURI;
 
+    // signer
+    address private _signer;
+
+    // series max supplies
+    mapping(uint256 => uint256) internal _seriesMaxSupplies;
+
+    // series total supplies
+    mapping(uint256 => uint256) internal _seriesTotalSupplies;
+
+    // total supply
+    uint256 private _totalSupply;
+
+    // all artworks
+    mapping(uint256 => Artwork) internal _allArtworks;
+
     constructor(
         string memory name_,
         string memory symbol_,
-        bool isBurnable_,
-        bool isBridgeable_
+        address signer_,
+        bool burnable_,
+        bool bridgeable_,
+        uint256[] memory seriesIds_,
+        uint256[] memory seriesMaxSupplies_
     ) ERC721(name_, symbol_) {
-        isBurnable = isBurnable_;
-        isBridgeable = isBridgeable_;
+        // validations
+        require(
+            bytes(name_).length > 0,
+            "FeralfileExhibitionV4: name_ is empty"
+        );
+        require(
+            bytes(symbol_).length > 0,
+            "FeralfileExhibitionV4: symbol_ is empty"
+        );
+        require(
+            signer_ != address(0),
+            "FeralfileExhibitionV4: signer_ is zero address"
+        );
+        require(
+            seriesIds_.length > 0,
+            "FeralfileExhibitionV4: seriesIds_ is empty"
+        );
+        require(
+            seriesMaxSupplies_.length > 0,
+            "FeralfileExhibitionV4: _seriesMaxSupplies is empty"
+        );
+        require(
+            seriesIds_.length == seriesMaxSupplies_.length,
+            "FeralfileExhibitionV4: seriesMaxSupplies_ and seriesIds_ lengths are not the same"
+        );
+
+        _signer = signer_;
+        _burnable = burnable_;
+        _bridgeable = bridgeable_;
+
+        // initialize max supply map
+        for (uint256 i = 0; i < seriesIds_.length; i++) {
+            _seriesMaxSupplies[seriesIds_[i]] = seriesMaxSupplies_[i];
+            require(
+                seriesMaxSupplies_[i] > 0,
+                "FeralfileExhibitionV4: zero max supply"
+            );
+        }
     }
 
     function supportsInterface(
         bytes4 interfaceId
-    ) public view virtual override(ERC721Enumerable, IERC165) returns (bool) {
-        return
-            interfaceId == type(IERC721Enumerable).interfaceId ||
-            super.supportsInterface(interfaceId);
+    ) public view virtual override(ERC721) returns (bool) {
+        return super.supportsInterface(interfaceId);
     }
 
-    /// @notice Return a count of series registered in this exhibition
-    function totalSeries() public view virtual returns (uint256) {}
+    function mintable() public view virtual returns (bool) {
+        return _mintable;
+    }
+
+    function setMintable(bool mintable_) external virtual onlyAuthorized {
```

Comment:
We should not make 
- setMintable
- setBurnable
- setBridgeable

be changed directly. It means FF can mint anything anytime if we want.
For open edition, I plan not do it over the new contract. And just make this a pure exhibition contract. So I think we only need to disable it when a sale starts and leave it false forever.

Reply:
@lemonlatte I just want to make it more flexible. If we don't concern about the security risks, why dont we keep them?

Reply:
When people look into the contract, they know that we CAN do this. If the policy is not do, it would be better we let people see this from the contract instead of leave some fuzzy space for us to do in the future.